### ISSUE TITLE
Local and third party packages are being sorted together

### ISSUE THREAD

**Issue by Siecje** [timothycrosley/isort#1104]
I am using `develop` but local imports and third party imports are being sorted together.

`~/.isort.cfg`
```
[settings]
known_third_party=gcutilities
force_sort_within_sections=1
no_lines_before=LOCALFOLDER
order_by_type=False
```

```
cody@ ~/D/why> tree
.
├── baz.py
└── Platform
    ├── foo.py
    └── __init__.py
cody@ ~/D/why> cat Platform/foo.py 
bar = 1

cody@ ~/D/why> cat baz.py 
from rest_framework import serializers

from Platform.foo import bar

```

```
cody@ ~/D/why> isort baz.py 
Fixing /home/cody/Desktop/why/baz.py
cody@ ~/D/why> cat baz.py 
from Platform.foo import bar
from rest_framework import serializers
```

**Comment by toumorokoshi** [comment]
I've noticed this as well. I don't see any clear documentation on isort's sort inference strategy, but isort is effectively looking to see if a package is installed to determine if it's first or third party.

For example, if I don't install flask into my virtualenv, it's considered a first party and sorted with the rest.

If I do install flask, it's considered a third party. 

It might be better in general for isort to have an uncategorized section. But at least documenting this behavior somewhere easily accessible would have saved me a lot of time.

**Comment by toumorokoshi** [comment]
Also it looks like #1071 is a good solution for this.

**Comment by timothycrosley** [comment]
This should be fixed for the 5.0.0 release in spirit via this ticket: https://github.com/timothycrosley/isort/issues/1147
